packages/cloud_firestore/lib/src/document_reference.dart

```diff
@@ -71,11 +71,12 @@ class DocumentReference {
   /// Reads the document referenced by this [DocumentReference].
   ///
   /// If no document exists, the read will return null.
-  Future<DocumentSnapshot> get() async {
+  /// Valid options for source are 'server', 'cache' or '' which is default
+  Future<DocumentSnapshot> get({String source = ''}) async {
```

Comment:
I think that `source` should have type `enum FirestoreSource` (which you can define in this file). Although the JavaScript SDK uses a String type it is IMO more idiomatic in Dart to use an enum.

Reply:
Hey @collinjackson I have added enum FirestoreSource to the source.